irohad/ordering/impl/ordering_gate_cache/on_demand_cache.hpp

```diff
@@ -29,9 +36,11 @@ namespace iroha {
         virtual const BatchesSetType &tail() const override;
 
        private:
+        const uint64_t max_cache_size_;
         mutable std::shared_timed_mutex mutex_;
-        using BatchesQueueType = boost::circular_buffer<BatchesSetType>;
-        BatchesQueueType circ_buffer{3, BatchesSetType{}};
+        using CacheElementType = std::pair<uint64_t, BatchesSetType>;
```

Comment:
Please add a comment that pair.first is the cumulative number of transactions in stored batches.